mismi-s3/test/Test/IO/Mismi/S3/Commands.hs

```diff
@@ -118,6 +118,34 @@ prop_size_recursively d = testAWS $ do
   r <- sizeRecursively (a { key = dirname $ key a })
   pure $ r === [Sized (fromIntegral . BS.length $ T.encodeUtf8 d) a]
 
+prop_join =
+  forAll arbitrary $ \bs -> (BS.length bs /= 0) ==> testAWS $ do
+    a <- newAddress
+    b <- newAddress
+    c <- newAddress
+    f <- newFilePath
+    let
+      s = f </> T.unpack "fred"
+      d = f </> T.unpack "down"
+    liftIO $ withFile s WriteMode $ \h ->
+      replicateM_ 1000 (LBS.hPut h (LBS.fromChunks . return $ (BS.concat . L.replicate 10000 $ bs)))
```

Comment:
Is a bit weird to convert a single chunk to a lazy bytestring just to write it out immediately, maybe this should just be:
```hs
replicateM_ 1000 (BS.hPut h . BS.concat $ L.replicate 10000 bs)
```
or even
```hs
let
  bs10k = BS.concat $ L.replicate 10000 bs
replicateM_ 1000 (BS.hPut h bs10k)
```
(if it's taking any amount of time atm)